packages/node/src/core/coordinator/calls/disaggregation.ts

```diff
@@ -51,34 +51,42 @@ function mapApiCalls(
   return [logs, apiCallWithResponses];
 }
 
-function mapProviderState(
+function mapEVMProviderState(
   state: ProviderState<EVMProviderState>,
   aggregatedApiCallsById: AggregatedApiCallsById
 ): LogsData<ProviderState<EVMProviderState>> {
-  const initialState: WalletDataWithLogs = { logs: [], walletDataByIndex: {} };
+  const requestsByRequesterIndex = grouping.groupRequestsByRequesterIndex(state.requests);
```

Comment:
Yeah for nonces. In this case you've highlighted, requests are grouped because I need to iterate through all of the API calls and then apply the response value/error from the `AggregatedApiCall`.

Looking at it again now, it could probably be simplified by just looping over all API calls first then grouping

Reply:
Actually it doesn't need to be grouped at all here 👍🏻

Reply:
Okay